libsolidity/analysis/DeclarationContainer.h

```diff
@@ -57,7 +56,12 @@ class DeclarationContainer
 	bool registerDeclaration(Declaration const& _declaration, ASTString const* _name, langutil::SourceLocation const* _location, bool _invisible, bool _update);
 	bool registerDeclaration(Declaration const& _declaration, bool _invisible, bool _update);
 
-	std::vector<Declaration const*> resolveName(ASTString const& _name, bool _recursive = false, bool _alsoInvisible = false) const;
+	std::vector<Declaration const*> resolveName(
+		ASTString const& _name,
+		bool _recursive = false,
+		bool _alsoInvisible = false,
+		bool _onlyVisibleAsUnqualifiedNames = false
+	) const;
```

Comment:
It would be nice to add a comment about the parameters. Also maybe something like `_onlyVisibleAsUnqualifiedNames` is applicable only to `UserDefinedTypeName` (is that true?)

Reply:
Right. The function has no docstring. I'll add one.

> `_onlyVisibleAsUnqualifiedNames` is applicable only to `UserDefinedTypeName` (is that true?)

I think it's true unless we have built-in type names that are not keywords.
Well, I guess would not hurt to add a test that tries to shadow built-in names and to check that.

Reply:
Test added. All builtin types I could think of cause parser errors when used as parameter names.